builtin/providers/aws/resource_aws_route.go

```diff
@@ -0,0 +1,337 @@
+				Computed: true,
+			},
+
+			"gateway_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"instance_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"instance_owner_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"network_interface_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"origin": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"state": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"route_table_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"vpc_peering_connection_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsRouteCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+	var numTargets int
+	var setTarget string
+	allowedTargets := []string{
+		"gateway_id",
+		"instance_id",
+		"network_interface_id",
+		"vpc_peering_connection_id",
+	}
+
+	// Check if more than 1 target is specified
+	for _, target := range allowedTargets {
+		if len(d.Get(target).(string)) > 0 {
+			numTargets++
+			setTarget = target
+		}
+	}
+
+	if numTargets > 1 {
+		fmt.Errorf("Error: more than 1 target specified. Only 1 of gateway_id" +
+			"instance_id, network_interface_id, route_table_id or" +
+			"vpc_peering_connection_id is allowed.")
+	}
+
+	createOpts := &ec2.CreateRouteInput{}
+	// Formulate CreateRouteInput based on the target type
+	switch setTarget {
+	case "gateway_id":
+		createOpts = &ec2.CreateRouteInput{
+			RouteTableId:         aws.String(d.Get("route_table_id").(string)),
+			DestinationCidrBlock: aws.String(d.Get("destination_cidr_block").(string)),
+			GatewayId:            aws.String(d.Get("gateway_id").(string)),
+		}
+	case "instance_id":
+		createOpts = &ec2.CreateRouteInput{
+			RouteTableId:         aws.String(d.Get("route_table_id").(string)),
+			DestinationCidrBlock: aws.String(d.Get("destination_cidr_block").(string)),
+			InstanceId:           aws.String(d.Get("instance_id").(string)),
+		}
+	case "network_interface_id":
+		createOpts = &ec2.CreateRouteInput{
+			RouteTableId:         aws.String(d.Get("route_table_id").(string)),
+			DestinationCidrBlock: aws.String(d.Get("destination_cidr_block").(string)),
+			NetworkInterfaceId:   aws.String(d.Get("network_interface_id").(string)),
+		}
+	case "vpc_peering_connection_id":
+		createOpts = &ec2.CreateRouteInput{
+			RouteTableId:           aws.String(d.Get("route_table_id").(string)),
+			DestinationCidrBlock:   aws.String(d.Get("destination_cidr_block").(string)),
+			VpcPeeringConnectionId: aws.String(d.Get("vpc_peering_connection_id").(string)),
+		}
+	default:
+		fmt.Errorf("Error: invalid target type specified.")
+	}
+	log.Printf("[DEBUG] Route create config: %s", awsutil.Prettify(createOpts))
```

Comment:
nitpick: `awsutil.Prettify` isn't necessary, the SDK implements GoStringer so you can just do `config: %s", createOpts)`